Updated adapter to support Rain.fi v2

solwhitey opened the pull request:
Hello, we updated our code to support our v2 protocol which involves two programs:
- Bank program (handles liquidity from pools)
- DeFi Lending program (handles loans)

Changes:
- Updated TVL calculation to include both bank liquidity and DeFi loan collateral
- Added new IDL files for both programs

The migration of user funds to these new programs has been completed, and this adapter reflects the new structure.

g1nt0ki commented on projects/rain/index.js:
```diff
@@ -1,45 +1,50 @@
-  for (const pool of pools)
-    api.add(pool.account.currency.toString(), pool.account.borrowedAmount)
-}
+tvl()
```
please remove this line

g1nt0ki commented on projects/rain/defi-idl.js:
```diff
@@ -0,0 +1,275 @@
+                      "fields": [
+                          {
+                              "defined": {
+                                  "name": "RequestLoan"
```
thanks for the PR, but the definition is missing for this

g1nt0ki reviewed (changes requested):
hi, I refactored the code a bit and added back the borrowed, can you doublecheck the code? the tvl feels wrong.

solwhitey commented:
> hi, I refactored the code a bit and added back the borrowed, can you doublecheck the code? the tvl feels wrong.

Hi, thank you, I fixed it, numbers should now be better
